flink-streaming-java/src/test/java/org/apache/flink/streaming/util/StreamTaskUtil.java

```diff
@@ -29,9 +30,10 @@
  */
 public class StreamTaskUtil {
 
-	public static void waitTaskIsRunning(StreamTask<?, ?> task, CompletableFuture<Void> taskInvocation) throws InterruptedException {
+	public static void waitTaskIsRunning(StreamTask<?, ?> task, CompletableFuture<Void> taskInvocation) throws InterruptedException, ExecutionException {
 		while (!task.isRunning()) {
 			if (taskInvocation.isDone()) {
+				taskInvocation.get();
```

Comment:
An alternative way to propagate the exception would be:
```
				taskInvocation.exceptionally(e -> {
					throw new AssertionError("Task has stopped", e);
				});
```

It depends if you want the test to fail through assertions or through unexpected exception.

Reply:
I'd prefer not to wrap it as it's more straightforward to me both in code and in the logs.